shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm

```diff
@@ -67,10 +67,18 @@ - (void)beginResize:(CGSize)size notify:(dispatch_block_t)notify {
 
   _waiting = YES;
 
-  _condBlockRequestCommit.wait(lock, [&] { return _pendingCommit; });
+  const int wait_time_millis = 500;
+  bool request_commit_success = _condBlockRequestCommit.wait_for(
+      lock, std::chrono::milliseconds(wait_time_millis), [&] { return _pendingCommit; });
+  if (!request_commit_success) {
+    NSLog(@"Commit wasn't requested after %d ms, no frame was requested."
```

Comment:
This seems really bad; please file an issue and add a comment here referencing that issue so it's clear that this code shouldn't exist and where the underlying issue is tracked.